src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementFhirImportService.cs

```diff
@@ -116,7 +116,7 @@ private static IEnumerable<IMeasurementGroup> ParseEventData(IEnumerable<IEventM
                 .Select(g =>
                 {
                     var measurements = g.ToList();
-                    _ = CalculateMetricsAsync(measurements, log).ConfigureAwait(false);
+                    _ = CalculateMetricsAsync(measurements, log, data.First().PartitionId).ConfigureAwait(false);

```

Comment:
>data [](http://example.com/codeflow?start=65&length=4)

This will result in multile iterations over the IEnumerable.  

I see three possible alternatives.

1. Prior to the Linq query do the first on the data element and store the partition in a variable and use that in the call.  You still have two iterations over the data IEnumerable but at least you aren't incurring the cost per group. This is my least favorite.
2. Instead you could update the CalculateMetricsAsync and order the Measurement and Measurement group count after you iterate through the measurement collection and set a variable for the partition in the for loop.  THe last value is what you use when you emit the count metrics afterwards.
3. You already have a list of Measurements in the var measurements.  Grab the partition from measurements[0].  This should be the most efficient method.